src/core/config/descriptors/config_descriptor.h

```diff
@@ -55,11 +57,18 @@ class config_descriptor {
      */
     std::type_index get_parent_expected_type(const std::string &key) const;
 
+    /**
+     * @brief Gets the parameter map, allowing external users to iterate 
+     *        over all parameters
+     * @return The parameter map
+     */
+    const parameter_map_t& get_parameter_map() const;
+
 private:
     /**
      * @brief Map from parameter name to its descriptor
      */
-    std::map<std::string, parameter_descriptor> parameter_map;
+     parameter_map_t parameter_map;
```

Comment:
run clang-format on all the files. Hope to have pre-commit PR merged soon so you should never worry about this again.